src/libraries/System.Text.Json/tests/Serialization/PolymorphicTests.cs

```diff
@@ -9,35 +9,62 @@
 
 namespace System.Text.Json.Serialization.Tests
 {
-    public static class PolymorphicTests
+    public class PolymorphicTests_String : PolymorphicTests
     {
+        public PolymorphicTests_String() : base(SerializationWrapper.StringSerializer) { }
+    }
+
+    public class PolymorphicTests_Stream : PolymorphicTests
+    {
+        public PolymorphicTests_Stream() : base(SerializationWrapper.StreamSerializer) { }
+    }
+
+    public class PolymorphicTests_StreamWithSmallBuffer : PolymorphicTests
+    {
+        public PolymorphicTests_StreamWithSmallBuffer() : base(SerializationWrapper.StreamSerializerWithSmallBuffer) { }
+    }
+
+    public class PolymorphicTests_Writer : PolymorphicTests
+    {
+        public PolymorphicTests_Writer() : base(SerializationWrapper.WriterSerializer) { }
+    }
+
+    public abstract class PolymorphicTests
+    {
+        private SerializationWrapper Serializer { get; }
```

Comment:
nit: use a more verbose name rather than the general `Serializer` (helps with future find/replace refactorings).

Something like: `CustomAllSerializer`

Reply:
Maybe `SerializeMethodOverload`.

Reply:
You can still search for "Serializer." via whole word, same case. i.e. it doesn't collide with "JsonSerializer."

Reply:
I think the tests look good like this (maybe a bit confusing for first 5 seconds however)
```cs
json = Serializer.Serialize<object>(array);
```
versus
```cs
json = CustomAllSerializer.Serialize<object>(array);
```
or
```cs
json = Serializer.SerializeMethodOverload<object>(array);
```